pyccel/codegen/printing/fcode.py

```diff
@@ -694,21 +695,16 @@ def _print_NumpyWhere(self, expr):
     def _print_NumpyArray(self, expr):
         """Fortran print."""
 
-        # Always transpose indices because Numpy initial values are given with
-        # row-major ordering, while Fortran initial values are column-major
-        shape = expr.shape[::-1]
-
         # Construct right-hand-side code
-        if expr.rank > 1:
-            arg = functools.reduce(operator.concat, expr.arg)
-            rhs_code = 'reshape({array}, {shape})'.format(
-                    array=self._print(arg), shape=self._print(Tuple(*shape)))
-        else:
-            rhs_code = self._print(expr.arg)
-
+        rhs_code = self._print(expr.arg)
         # If Numpy array is stored with column-major ordering, transpose values
+        # use reshape with order for rank > 2
         if expr.order == 'F' and expr.rank > 1:
```

Comment:
I don't see the `else`. Is C ordering handled?

Reply:
yes it's you don't need the else